src/pages/CDP.js

```diff
@@ -114,21 +117,59 @@ const CdpViewCard = ({ title, rows, isAction }) => {
   );
 };
 
-function CDPView({ cdpId }) {
+function CDPView({ cdpTypeSlug, getIlk }) {
   const { maker } = useMaker();
 
   // TODO cdpTypeSlug should become `id` or we should have both cdpTypeSlug AND id.
   const [cdpState, setCDPState] = useState(null);
+  const [collateralVal, setCollateralVal] = useState('');
+  const [debtVal, setDebtVal] = useState('');
+  const [ilk, setIlk] = useState(null);
 
   useEffect(() => {
     (async () => {
       const cdpManager = maker.service('mcd:cdpManager');
-      const cdp = await cdpManager.getCdp(cdpId);
+      const cdp = await cdpManager.getCdp(parseInt(cdpTypeSlug));
       setCDPState(cdp);
+      setIlk(getIlk(cdp.ilk));
+      setCollateralVal(await cdp.getCollateralValue());
+      setDebtVal(await cdp.getDebtValue());
     })();
   }, [cdpId, maker]);
 
   console.log('CDP state to be rendered on the page:', cdpState);
+  window.cdpState = cdpState;
+  let { liquidationPrice, collateralPrice, liquidationPenalty } = ['', 1, ''];
```

Comment:
I'm guessing you're using this pattern so that the page has a placeholder state to render with?

perhaps a better approach would be to do 
```js
if ([cdpState. collateralVal, debtVal, ilk].includes(false)) {
  return <LoadingLayout />
}
//... render once all is available
```
that way avoiding rewriting variables and having a nice loading state while fetching data, later on we should also consider using `Suspense`.

Reply:
agreed this a better pattern I'll update it